support/cas-server-support-aup-redis/src/main/java/org/apereo/cas/config/CasAcceptableUsagePolicyRedisConfiguration.java

```diff
@@ -27,7 +27,7 @@
  */
 @Configuration("casAcceptableUsagePolicyRedisConfiguration")
 @EnableConfigurationProperties(CasConfigurationProperties.class)
-@ConditionalOnProperty(prefix = "cas.acceptable-usage-policy.core", name = "enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnProperty(prefix = "cas.acceptable-usage-policy.core", name = { "enabled", "redis.enabled" })
```

Comment:
I am not sure I follow. The second bit here translates to `cas.acceptable-usage-policy.core.redis.enabled`? Perhaps an easier way would be to just declare two annotations each for separate conditions?

Reply:
From what I could tell the prefix and the name are combined with a `.` between them so I think this would work... I could try something like this:

```
@ConditionalOnExpression(value = "${casProperties.acceptable-usage-policy.core.enabled} and ${casProperties.acceptable-usage-policy.core.redis.enabled}")
```

Reply:
```
@ConditionalOnExpression(value = "${cas.acceptable-usage-policy.core.enabled:true} and ${cas.acceptable-usage-policy.core.redis.enabled:true}")
```
Disables classes if aup globally disabled or if redis module disabled, both are enabled by default. 

```
@ConditionalOnExpression(value = "${cas.authn.saml-idp.metadata.redis.idp-metadata-enabled:false} and ${cas.authn.saml-idp.metadata.redis.enabled:true}")
```
The SAML extension of the redis properties object adds an existing idp-metadata-enabled flag which defaults to false and the base class has the new enabled property. They are both needed b/c one could be using redis for SAML service provider metadata but not want to use it for IDP metadata. 

I added some tests that I think show this is working, although it's easy to make typos on these expressions.